src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -57,13 +57,16 @@ private sealed class Parser
             /// <summary>
             /// Type information for member types in input object graphs.
             /// </summary>
-            private readonly Dictionary<Type, TypeMetadata> _typeMetadataCache = new();
+            private readonly Dictionary<Type, TypeGenerationSpec> _typeGenerationSpecCache = new();
 
             public Parser(Compilation compilation)
             {
                 _compilation = compilation;
                 _metadataLoadContext = new MetadataLoadContextInternal(compilation);
 
+                TypeExtensions.NullableOfTType = _metadataLoadContext.Resolve(typeof(Nullable<>));
```

Comment:
This isn't a great coding pattern - setting a static variable here, and reading it in `TypeGenerationSpec`. It makes the code fragile because it always assumes this gets called first.

Reply:
It bothered me as well - I've refactored the code to not use this pattern.